src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/CodeActionOptions.cs

```diff
@@ -31,7 +31,11 @@ namespace Microsoft.CodeAnalysis.CodeActions
     [DataContract]
     internal sealed record class CodeActionOptions
     {
+#if CODE_STYLE
         public static readonly CodeActionOptionsProvider DefaultProvider = new DelegatingCodeActionOptionsProvider(GetDefault);
+#else
+        public static readonly CodeActionOptionsProvider DefaultProvider = new DelegatingCodeActionOptionsProvider(static ls => GetDefault(ls.ProjectServices));
```

Comment:
Why is this changing here? Does DelegatingCodeActionOptionsProvider also need to be updated or something? (this feels either like it's a temporary hack or some strange layering compilation, but in either case probably needs a comment?)

Reply:
Yes.  this needs to be updated.  And it *can* be once this API is public.  We're limited here as this is linked into the code-style layer.  So come thursday this will be fine and is part of all the work i'm expecting to do.